shell/packages/sandstorm-db/db.js

```diff
@@ -587,14 +638,14 @@ isUserExcessivelyOverQuota = function (user) {
 
   if (!Meteor.settings.public.quotaEnabled || user.isAdmin) return false;
 
-  var plan = Plans.findOne(user.plan || "free");
+  var quota = getUserQuota(user);
 
-  if (plan.grains < Infinity) {
-    var count = Grains.find({userId: user._id}, {fields: {}, limit: plan.grains * 2}).count();
-    if (count >= plan.grains * 2) return "outOfGrains";
+  if (quota.grains < Infinity) {
```

Comment:
This was slightly confusing to me at first, since I'm not all that familiar with the rules for comparing numbers to null/undefined. It might be better to make your check explicitly be `quota.grains !== undefined`.